doc/debugger.rst

```diff
@@ -922,6 +935,8 @@ Error codes
 +--------+------------------------------------------------------------------+
 | 0x03   | Not found (e.g. invalid breakpoint index)                        |
 +--------+------------------------------------------------------------------+
+| 0x04   | AppRequest error (i.e. request callback returned -1)             |
```

Comment:
Sorry for going back and forth, there are somewhat competing concerns here :) But application error sounds like a good compromise, it's at the same pretty generic level and maybe it could affect how the debug client displays an error for example.